components/style/properties/helpers.mako.rs

```diff
@@ -607,3 +611,70 @@
         }
     </%self:shorthand>
 </%def>
+
+<%def name="logical_setter_helper(name)">
+    <%
+        side = None
+        size = None
+        maybe_side = [s for s in LOGICAL_SIDES if s in name]
+        maybe_size = [s for s in LOGICAL_SIZES if s in name]
+        if len(maybe_side) == 1:
+            side = maybe_side[0]
+        elif len(maybe_size) == 1:
+            size = maybe_size[0]
```

Comment:
Probably just `assert size is not None or side is not None`, and just use `else` below?